pyiron/vasp/base.py

```diff
@@ -2135,15 +2178,52 @@ def __init__(self, input_file_name=None, table_name="kpoints"):
             val_only=True,
             comment_char="!",
         )
+        self._trace = None
+
+    def _set_trace(self, trace):
+        """
+        Sets high symmetry points names of k-points trace (line mode only)
+
+        Args:
+            trace (list): new trace
+        """
+        self._trace = trace
 
-    def set(self, method=None, size_of_mesh=None, shift=None):
+    def _get_trace(self):
+        """
+        Returns high symmetry points name of k-points trace (Line mode only)
+
+        Returns:
+            list: trace values
+        """
+        return self._trace
+
+    def set(self, method=None, size_of_mesh=None, shift=None, n_trace=None, trace_coord=None):
         """
         Sets appropriate tags and values in the KPOINTS file
         Args:
             method (str): Type of meshing scheme (Gamma Point, MP, Manual or Line)
             size_of_mesh (list/numpy.ndarray): List of size 1x3 specifying the required mesh size
             shift (list): List of size 1x3 specifying the user defined shift from the Gamma point
+            n_trace (int): Number of points per trace for line mode
+            trace_coord (list): coordinates of k-points trace in VASP KPOINTS format
         """
+        if n_trace is not None:
+            if self._trace is None or trace_coord is None:
+                raise ValueError("trace have to be defined")
+
+            self.set_value(line=1, val=n_trace)
```

Comment:
The Kpoints file is the only one that uses `set_value()` instead of `self[1] = n_trace` - @sudarsan-surendralal Is there a specific reason for this?

Reply:
This would not work since in this case, the line number does not exist till now
```
from pyiron import Project
pr = Project("setval_demo")
job = pr.create_job(pr.job_type.Vasp, "setval_demo")
print(job.input.kpoints._dataset)
job.input.kpoints[5] = "5 5 5"

>> ---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-5-2614e019c818> in <module>
      3 job = pr.create_job(pr.job_type.Vasp, "setval_demo")
      4 print(job.input.kpoints._dataset)
----> 5 job.input.kpoints[5] = "5 5 5"

/cmmc/u/chandu/programs/pyiron_mpie/pyiron/pyiron/base/generic/parameters.py in __setitem__(self, key, value)
    640             if self.read_only and self._dataset["Value"][key] != value:
    641                 self._read_only_error()
--> 642             self._dataset["Value"][key] = value
    643         else:
    644             self.set(**{key: value})

IndexError: list assignment index out of range
```
Therefore, we have to stick to `set_value`